Debugger/src/Connection/Grpc.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Debugger\Connection;
+
+use Google\ApiCore\Serializer;
+use Google\Cloud\Core\GrpcRequestWrapper;
+use Google\Cloud\Core\GrpcTrait;
+use Google\Cloud\Debugger\DebuggerClient;
+use Google\Cloud\Debugger\V2\Breakpoint;
+use Google\Cloud\Debugger\V2\Debuggee;
+use Google\Cloud\Debugger\V2\Controller2Client;
+use Google\Cloud\Debugger\V2\Debugger2Client;
+
+/**
+ * Implementation of the
+ * [Google Debugger gRPC API](https://cloud.google.com/debugger/docs/).
+ */
+class Grpc implements ConnectionInterface
+{
+    use GrpcTrait;
+
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    /**
+     * @var Controller2Client
+     */
+    private $controllerClient;
+
+    /**
+     * @var Debugger2Client
+     */
+    private $debuggerClient;
+
+    /**
+     * @param array $config
+     */
+    public function __construct(array $config = [])
+    {
+        $this->serializer = new Serializer([], [
+            'google.protobuf.Timestamp' => function ($v) {
+                return $this->formatTimestampFromApi($v);
+            },
+            'google.protobuf.Int32Value' => function ($v) {
+                return $this->flattenValue($v);
+            }
+        ], [], [
+            'google.protobuf.Timestamp' => function ($v) {
+                return $this->formatTimestampForApi($v);
+            },
+            'google.protobuf.Int32Value' => function ($v) {
+                return [
+                    'value' => $v
+                ];
+            }
+        ]);
+        $config['serializer'] = $this->serializer;
```

Comment:
$this->serializer always wins. Does it make sense to allow users to specify their own serializer?

Reply:
My first thought is to keep the serializer a private concept to the library. The goal of it existing was to create parity with the manual REST transport, and it seems like users creating their own implementations could be an easy way to introduce mistakes.

I am open to the concept, though, what value do you see a custom serializer providing?

Reply:
I don't have a specific use case. I'm totally OK we just keep it private for now :)
If we add the documentation, it's more clear.